pkg/operator/ceph/object/spec.go

```diff
@@ -309,6 +310,23 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) v1.Container {
 					c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.VaultSecretEngineKey]),
 			)
 		}
+		if c.store.Spec.Security.KeyManagementService.IsTLSEnabled() &&
+			c.clusterInfo.CephVersion.IsAtLeast(cephver.CephVersion{Major: 16, Minor: 2, Extra: 5}) {
+			container.Args = append(container.Args,
+				cephconfig.NewFlag("rgw crypt vault verify ssl", "true"))
+			if kms.GetParam(c.store.Spec.Security.KeyManagementService.ConnectionDetails, api.EnvVaultClientCert) != "" {
+				container.Args = append(container.Args,
+					cephconfig.NewFlag("rgw crypt vault ssl clientcert", path.Join(c.DataPathMap.ContainerDataDir, kms.VaultCertFileName)))
+			}
+			if kms.GetParam(c.store.Spec.Security.KeyManagementService.ConnectionDetails, api.EnvVaultClientKey) != "" {
+				container.Args = append(container.Args,
+					cephconfig.NewFlag("rgw crypt vault ssl clientkey", path.Join(c.DataPathMap.ContainerDataDir, kms.VaultKeyFileName)))
+			}
+			if kms.GetParam(c.store.Spec.Security.KeyManagementService.ConnectionDetails, api.EnvVaultCACert) != "" {
+				container.Args = append(container.Args,
+					cephconfig.NewFlag("rgw crypt vault ssl cacert", path.Join(c.DataPathMap.ContainerDataDir, kms.VaultCAFileName)))
+			}
+		}
```

Comment:
Anything placed in the ContainerDataDir could be persisted to the host, and it is for information only for the specific Ceph daemon. Similarly to how Vault is configured for OSDs, there should be a completely separate volume mount for these keys and certs.

You should look to see if you can reuse any code found in `pkg/daemon/ceph/osd/kms/volumes.go`. If so, perhaps we should have move reusable bits to `pkg/daemon/ceph/kms` or someplace similar. What do you think @leseb ?

Reply:
Sorry I was not aware of that. Another option is to add these in "/etc/ceph" code path, but all the files owned in that location was by `root`. Here the file needs to be owned by `ceph:ceph` so that the RGW process can access it and it is done via init container

Reply:
It looks like Seb configured the KMS for OSDs to use `/etc/vault`

Reply:
@BlaineEXE : By default, secrets are mounted read-only files with `root:root`and RGW needs this to be `ceph` owned file. I cannot chown on this mounted file or add FSGroup to `ceph` podspec(in openshift this value will be set by default). So I am copying these files to `containerdir` and then perform chown via init container(refer `setupVaultTokenFile`). By default file is mounted in the same location `/etc/vault`.

Reply:
I think this is probably also a requirement of KMS on OSDs, but @leseb would know for sure. Having KMS handling/configuration that is shared between rook-ceph components will make things easier to maintain, debug, and unit test.

Reply:
@BlaineEXE : I cannot create this new directory in `/etc/ceph` or `/etc/vault` because they are already mounted in the rgw pod as read-only secret. If I try to create the directory via init container, it won't be persisted for the main rgw container, so approach to save it  `ContainerDataDir`. Another way to resolve this issue is in ceph, remove restrictions on the token file for the vault(currently it restricts to have permission to only to the owner of the file, not even `read` permission to group/others) but not sure whether it will acceptable from the ceph community

Reply:
This content **cannot** go into `ContainerDataDir` for the reasons I explained. It is wholly untrue that this is the only feasible approach. Examine the OSD KMS code that Seb wrote (that this PR should reuse as much as possible), and see that `/etc/vault` is mounted from an `emptyDir`. There is no need to make changes to Ceph.